src/main/java/teammates/ui/controller/FeedbackSubmissionEditSaveAction.java

```diff
@@ -234,6 +250,198 @@ protected ActionResult execute() throws EntityDoesNotExistException {
         return createSpecificRedirectResult();
     }
 
+    private void updateResponsesComments(Map<String, String> commentsToUpdateId,
+                                         Map<String, String> commentsToUpdateText,
+                                         Map<String, String> responseGiverMapForComments,
+                                         Map<String, String> responseRecipientMapForComments,
+                                         Map<String, String> questionIdsForComments) {
+        for (String commentIndex : commentsToUpdateId.keySet()) {
+            String showCommentTo =
+                    getRequestParamValue(Const.ParamsNames.RESPONSE_COMMENTS_SHOWCOMMENTSTO + commentIndex);
+            String showGiverNameTo =
+                    getRequestParamValue(Const.ParamsNames.RESPONSE_COMMENTS_SHOWGIVERTO + commentIndex);
+            String commentId = commentsToUpdateId.get(commentIndex);
+            String updatedCommentText = commentsToUpdateText.get(commentIndex);
+            FeedbackResponseAttributes response =
+                    logic.getFeedbackResponse(questionIdsForComments.get(commentIndex),
+                            responseGiverMapForComments.get(commentIndex),
+                            responseRecipientMapForComments.get(commentIndex));
+            String giverRole =
+                    getRequestParamValue(Const.ParamsNames.GIVER_ROLE + commentIndex);
+            updateResponseComment(showCommentTo, showGiverNameTo, commentId, response, updatedCommentText, giverRole);
+
+        }
+    }
+
+    private void updateResponseComment(String showCommentTo, String showGiverNameTo, String commentId,
+                                       FeedbackResponseAttributes response, String updatedCommentText,
+                                       String giverRole) {
+        FeedbackResponseCommentAttributes feedbackResponseComment = FeedbackResponseCommentAttributes
+                .builder(courseId, feedbackSessionName, response.giver, new Text(updatedCommentText))
+                .withCreatedAt(Instant.now())
+                .withGiverSection(response.giverSection)
+                .withReceiverSection(response.recipientSection)
+                .withGiverRole(giverRole)
+                .build();
+
+        feedbackResponseComment.setId(Long.parseLong(commentId));
+
+        //Edit visibility settings
+        feedbackResponseComment.showCommentTo = new ArrayList<>();
```

Comment:
DRY. You can create a (static) method in feedbackResponseComment to extract details from `requestParameters` like what we do for question details.

Reply:
Again, is student allowed to choose the visibility options?